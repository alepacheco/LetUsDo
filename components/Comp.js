function Home() {
  return <div>Imported</div>;
}

export default Home;