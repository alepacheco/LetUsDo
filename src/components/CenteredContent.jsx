import React from 'react';
import '../styles/components/CenteredContent.scss';

export const CenteredContent = ({ children, reverse }) => {
  return <div className={`centered-content ${reverse ? 'reverse' : ''}`}>{children}</div>;
};

export default CenteredContent;
