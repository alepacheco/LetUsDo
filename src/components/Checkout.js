import React from 'react';
import StripeCheckout from 'react-stripe-checkout';
import Button from 'react-bootstrap/Button';

export default class Checkout extends React.Component {
    constructor(props) {
        super(props);

        this.onToken = this.onToken.bind(this);
    }

    onToken(token) {
        fetch('/save-stripe-token', {
            method: 'POST',
            body: JSON.stringify(token),
        }).then(response => {
            response.json().then(data => {
                alert(`We are in business, ${data.email}`);
            });
        });
    }

    render() {
        return (
            <StripeCheckout
                name="We Do"
                description={this.props.description || 'Complete task'}
                panelLabel="Pay"
                amount={this.props.amount || 2000}
                currency="GBP"
                token={this.onToken}
                stripeKey="pk_test_TYooMQauvdEDq54NiTphI7jx"
            >
                <Button className="get-it-done-button">Get it done</Button>
            </StripeCheckout>
        )
    }
}