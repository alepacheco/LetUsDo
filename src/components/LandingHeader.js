import React from 'react';
import '../styles/components/LandingHeader.scss';
import Icon from '../static/images/undraw_process.svg';
import Form from 'react-bootstrap/Form';
import Button from 'react-bootstrap/Button';

const NewsletterForm = ({ }) => {
  return (
    <div className="newsletter-form">
      <div className="newsletter-email">
        <Form.Control type="email" placeholder="Enter email" />
        <div className="newsletter-substitle">
          We'll never share your email with anyone else.
            </div>
      </div>

      <Button variant="primary" type="submit" className="newsletter-button">
        Submit
      </Button>
    </div>
  )
};



export class LandingHeader extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <div className="landing-header">
        <div className="logo-text">
          LetUsDo.app
                </div>

        <div className="main-content">
          <h3>Get over with all the insudious task of your daily work.</h3>
          <h4>Get someone to do it for you for a flat fee.</h4>
          <p>Sign up for beta updates: </p>

          <NewsletterForm />
        </div>
        <div className="center-image">
          <Icon width="250" height="250" />
        </div>
      </div>
    );
  }
}

export default LandingHeader;
