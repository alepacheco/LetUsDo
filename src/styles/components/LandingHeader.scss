.landing-header {
  background-image: url('data:image/svg+xml;charset=utf8,%3Csvg%20xmlns=%22http://www.w3.org/2000/svg%22%20width=%221440%22%20height=%22661.1%22%20viewBox=%220%200%201440%20661.1%22%3E%3Cstyle%20type=%22text/css%22%3Ecircle,%20ellipse,%20line,%20path,%20polygon,%20polyline,%20rect,%20text%20%7B%20fill:%20%235c6ac4%20!important;%20%7D%3C/style%3E%3Cpath%20fill=%22%235c6ac4%22%20d=%22M1440%20488c-19.3%204.3-39.2%208.6-60%2013-174.2%2036.5-228.8%20288-467%2076.2-153.6-136.6-424-22.8-660.4-5.8-130.2%209.3-252.6-75-252.6-75V0h1440%22/%3E%3C/svg%3E');
  height: 662px;
  margin-bottom: -218px;

  margin-top: -50px;
  padding-top: 80px;

  @media only screen and (max-width: 1424px) {
    padding-top: 30px;
    margin-top: 0;
    height: 604px;
  }
  @media only screen and (max-width: 470px) {
    padding-top: 30px;
    margin-top: 0;
    height: 612px;
  }
  background-repeat: no-repeat;
  background-position: top center;
  background-position-x: center;
  background-position-y: -124px;

  background-size: cover;
  position: relative;

  .center-image {
    width: 250px;
    bottom: 0;
    position: absolute;

    left: 50%;
    -ms-transform: translate(-50%);
    transform: translate(-50%);
  }

  .logo-text {
    font-family: JelleeRoman, Sans-Serif;
    font-size: 24px;
    margin-left: 16px;
    margin-top: -8px;
  }

  .main-content {
    font-family: Quicksand, Sans-Serif;

    color: white;
    max-width: 450px;
    @media only screen and (max-width: 470px) {
      padding: 0 40px;
    }
    padding: 0 16px;

    margin: 88px auto 0 auto;
    font-size: 24px;
    text-align: center;
  }

  .newsletter-form {
    margin: 0 auto;
    width: 400px;
    display: flex;

    @media only screen and (max-width: 470px) {
      flex-flow: column;
      width: auto;
      margin: 0 36px;
    }
  }

  .newsletter-email {
    flex: 3;
  }

  .newsletter-button {
    flex: 0;
    height: fit-content;
    z-index: 2;
    margin-left: 8px;
    @media only screen and (max-width: 470px) {
      width: fit-content;
      margin: 0 auto;
    }
  }

  .newsletter-substitle {
    font-size: 12px;
    color: gainsboro;
  }

  .subtitle {
    font-weight: 300;
    color: rgba(235, 235, 235, 0.7);
  }
}
