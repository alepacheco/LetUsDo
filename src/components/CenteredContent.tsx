import React from 'react';
import 'src/styles/components/CenteredContent.scss';

type CenteredContentProps = {
  reverse?: boolean;
};

export const CenteredContent: React.FC<CenteredContentProps> = ({ children, reverse }) => {
  return <div className={`centered-content ${reverse ? 'reverse' : ''}`}>{children}</div>;
};

export default CenteredContent;
