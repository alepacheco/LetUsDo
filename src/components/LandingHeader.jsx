import React from 'react';
import '../styles/components/LandingHeader.scss';
import Form from 'react-bootstrap/Form';
import Button from 'react-bootstrap/Button';
import Icon from '../static/images/undraw_process.svg';

const NewsletterForm = ({}) => (
  <div className="newsletter-form">
    <div className="newsletter-email">
      <Form.Control type="email" placeholder="Enter email" />
      <div className="newsletter-substitle">We'll never share your email with anyone else.</div>
    </div>

    <Button variant="primary" type="submit" className="newsletter-button">
      Submit
    </Button>
  </div>
);

export const LandingHeader = () => (
  <div className="landing-header">
    <div className="logo-text">LetUsDo.app</div>
    <div className="main-content">
      We help you with the
      <b> tasks </b>
      that you don&apos;t feel like doing
    </div>
    <div className="center-image">
      <Icon width="250" height="250" />
    </div>
  </div>
);

export default LandingHeader;
