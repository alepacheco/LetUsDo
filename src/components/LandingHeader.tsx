import React from 'react';
import 'src/styles/components/LandingHeader.scss';

export const LandingHeader = () => (
  <div className="landing-header">
    <div className="logo-text">LetUsDo.app</div>
    <div className="main-content">
      <b>Delegate your tasks on-demand.</b>
      <div className="subtitle">
        Work clutters our minds. Choose what matters, delegate the rest.
      </div>
    </div>
  </div>
);

export default LandingHeader;
