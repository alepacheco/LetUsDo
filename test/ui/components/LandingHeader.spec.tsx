/* eslint-disable no-undef, import/no-unresolved */
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { LandingHeader } from 'components/LandingHeader';

jest.mock('src/static/images/undraw_process.svg', () => () => <div> img </div>);

describe('<LandingHeader />', () => {
  it('renders', () => {
    const wrapper = shallow(<LandingHeader />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
