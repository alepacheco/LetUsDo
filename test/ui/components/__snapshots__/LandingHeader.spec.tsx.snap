// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LandingHeader /> renders 1`] = `
<div
  className="landing-header"
>
  <div
    className="logo-text"
  >
    LetUsDo.app
  </div>
  <div
    className="main-content"
  >
    <b>
      Delegate your tasks on-demand.
    </b>
    <div
      className="subtitle"
    >
      Work clutters our minds. Choose what matters, delegate the rest.
    </div>
  </div>
</div>
`;
